Update gdal_rasterize.rst to optimize -optim (#11771)
OSGeo/gdal: doc/source/programs/gdal_rasterize.rst

.. option:: -optim {AUTO|VECTOR|RASTER}

    Force the algorithm used (results are identical). Raster mode

>     is used in most cases and  optimizes  read/write  operations.  The
>     vector  mode is useful with a large amount of input features and

Reviewer: ```suggestion
    is used in most cases and  optimizes read/write  operations.  The
    vector mode is useful with a large amount of input features and
```

    optimizes CPU use, provided that the input image is tiled.

>     Auto mode (the default) will chose the

Reviewer 2: output image

Reviewer: didn't understand your first suggestion ("-> that mode has to be used"). Fixes for other issues in https://github.com/OSGeo/gdal/pull/11775

Reviewer 2: It refers to the error in English in "That mode **have** to be used with tiled images..." but it is now gone anyway.

>     to be efficient. Auto mode (the default) will chose the

Reviewer 2: I believe you changed the meaning wrongly. Vector mode can be efficient only when the output raster is tiled.  Tiled output raster does not require vector mode.
I had a feeling that "decent" means here "not a huge number of input features" but it maybe I am wrong.

Author: I assure you that I have a huge stack of coins to flip when trying to guess the meaning of these. So would be ever grateful if someone who knows better than me steps in and words it better, please. Thanks!

>     optimizes CPU use.  Tiled images also need to use vector mode
>     to be efficient. Auto mode (the default) will chose the

Reviewer: ```suggestion
    optimizes CPU use, provided that the input image is tiled.
    Auto mode (the default) will chose the
```
